include/fluent-bit/flb_config.h

```diff
@@ -266,6 +267,8 @@ struct flb_config {
     int enable_hot_reload;
     int ensure_thread_safety_on_hot_reloading;
     unsigned int hot_reloaded_count;
+    unsigned int hot_reloaded_failures_count;
+    volatile sig_atomic_t bin_restarting;
```

Comment:
why this ?

Reply:
Originally this code was used to improve the reliability of the reload API so that it could be used in testing:

- `bin_restarting` was refactored out of the `src/fluent-bit.c` file to expose it to the reload API to better track the reload state
- hot_reloaded_failures_count was added to track failed reloads.

The `hot_reloaded_failures_count` is not crucial to this patch and can be removed.

Reply:
@edsiper should I drop `hot_reloaded_failures_count` and the relevant code?

Reply:
I meant: ` volatile sig_atomic_t ` why ?

Reply:
It is used across thread boundaries. Using `volatile sig_atomict_t` avoids using mutexes.

The alternative is to use a mutex or have differing implementations for signaling restarts within fluent-bit for Unix and windows.

Since the value is used as a flag I felt a mutex might be overblown and overly complex.

Reply:
can we rename `bin_restarting` to something like `service_is_restarting` ?

Reply:
I added a new function `flb_signal_reload` for signaling a reload from inside fluent-bit. In the future it can be refactored to use a better mechanism on windows.

I also renamed `bin_restarting` to `service_is_restarting`.